### ISSUE TITLE
Video titles (and perhaps descriptions) containing particular unicode/emojis/special characters will break the file move functions

### ISSUE THREAD

**Issue by JoryAnderson** [axcore/tartube#554]
Windows 10 latest updates
Tartube v2.4.370

Hi @axcore. Here are some videos where this happens a lot. Not all videos experience this issue, in particular it seems like this affects videos with certain special characters and emojis in the title or description.

[This one doesn't work](https://youtu.be/DeBCtwObnZk) (notice the abundance of special characters)
[Here's another video with an emoji in the title/description](https://youtu.be/UAQ_6u-gOKo)
[One more](https://youtu.be/dZc-3PT5Qyk)

[Here's a Beatles' song](https://youtu.be/UelDrZ1aFeY) with an ample description that will work. There's a hyphen in the title but that doesn't seem to break the thumbnail generation or write/move process.
[Here's Emoji by Brooklyn Queen.](https://youtu.be/SCSZb1Wl-8k) This one works too.

Here are my post-processing and file move/keep settings:
![image](https://github.com/axcore/tartube/assets/5385311/e23179fd-9904-40c8-8725-28bed8078b42)
![image](https://github.com/axcore/tartube/assets/5385311/beed1d11-8eb1-45d4-9c16-da4694b7f186)
![image](https://github.com/axcore/tartube/assets/5385311/f253036a-1dd3-458c-b596-a58d847a5a82)

Here is my resulting folders. Notice how The Beatles / Brooklyn Queen song is tidied up, while the emoji ones are not:
![image](https://github.com/axcore/tartube/assets/5385311/5ed69cf4-4556-4ecf-a589-c703a96489e5)
![image](https://github.com/axcore/tartube/assets/5385311/d7efcc53-8013-421a-996d-797fc9079eb8)
![image](https://github.com/axcore/tartube/assets/5385311/a66b9ca2-0a9b-4c91-b480-fded8d97bff1)

This also prevents the thumbnail from displaying on the UI:
![image](https://github.com/axcore/tartube/assets/5385311/48db5fbf-65dd-4b58-8db6-4ecb6b4d380c)

I downloaded these songs individually, as opposed to the whole folder. Later I decided to test downloading the whole folder. I cleaned out everything from the folder in File Explorer, then refreshed/checked/downloaded the folder from Tartube, and each time got the same result as downloading them individually. The only thing of note is that the Tartube UI doesn't display the emoji videos as downloaded (or display their size) in the Progress tab:
![image](https://github.com/axcore/tartube/assets/5385311/935a9009-036f-4fc5-9013-9b8a5cecb106)

This also prevents thumbnail conversion to jpg (in addition to file moving it to a subfolder). "Write the video's thumbnail to the same folder" will write the webp as expected.

_Originally posted by @JoryAnderson in https://github.com/axcore/tartube/issues/508#issuecomment-1572728860_

**Comment by JoryAnderson** [comment]
I decided to test a less egregious video (no unicode, basic special characters), and I got the same result. [Here's the video.](https://youtu.be/5vXk9M2RJFU). It has no description, the only special characters being"(/)" in the title. Attempting to delete the video via the context menu did not delete the videos from my filesystem.

**Comment by sergivonavi** [comment]
I have the same issue. After downloading a **playlist**, several videos are not recognized by Tartube, despite the fact that they exist in the destination folder (including their metadata and stuff).

Items have no description, etc.:
![tartube](https://github.com/axcore/tartube/assets/2625014/589f88ac-c627-4ab6-938b-15bb884dfc54)

All missing videos have the following special characters (one of them or both) in their original titles: `/ :`

After downloading they were changed to similar symbols (checked via fontspace.com) in the video file name:
- `/` (U+002F, forward slash) was changed to `⧸` (U+29F8);
- `:` (U+003A, colon) was changed to `：` (U+FF1A).

Other symbols (tested with `, ( ) &`) do not break the app.

Tried to download these videos separately (via "Add New Videos") and almost the same thing happened:
this time some metadata (description, duration, date) is shown, but not a thumbnail, comments and size.

`Right click on an item -> Special -> Download with name ->` inputting a name without `/` or `:` fixes the issue. But unfortunately this option is not available for videos in a playlist (Download with name is greyed out).

I also tried to rename and replace the file manually via `Right click on an item -> Show video -> Properties -> Set the file` but it doesn't work in a playlist (only works for a separately downloaded video):
![tartube2](https://github.com/axcore/tartube/assets/2625014/ad2d67ff-db91-451f-8d7d-b6560c16693f)

**Comment by axcore** [comment]
Ok, there seems to be two completely separate issues here. Let's deal with the forward slash character first, using @JoryAnderson 's [video](https://youtu.be/5vXk9M2RJFU).

yt-dlp is apparently not able to download the video, therefore Tartube can't handle it. Here's the output from a minimal download:

```
$ yt-dlp https://youtu.be/5vXk9M2RJFU
[youtube] Extracting URL: https://youtu.be/5vXk9M2RJFU
[youtube] 5vXk9M2RJFU: Downloading webpage
[youtube] 5vXk9M2RJFU: Downloading android player API JSON
[youtube] 5vXk9M2RJFU: Downloading MPD manifest
[info] 5vXk9M2RJFU: Downloading 1 format(s): 303+251
[dashsegments] Total fragments: 49
[download] Destination: 1v1 M24 intense fight(⧸) [5vXk9M2RJFU].f303.webm
[download] 100% of  102.07MiB in 00:00:28 at 3.57MiB/s
[dashsegments] Total fragments: 1
[download] Destination: 1v1 M24 intense fight(⧸) [5vXk9M2RJFU].f251.webm
[download] 100% of    3.06MiB in 00:00:00 at 3.48MiB/s
[Merger] Merging formats into "1v1 M24 intense fight(⧸) [5vXk9M2RJFU].webm"
ERROR: Postprocessing: Conversion failed!
```

Your next step is to mark this video as undownloadable at [yt-dlp's issues page](https://github.com/yt-dlp/yt-dlp/issues), noting that the same forward slash character is causing similar results with other videos.

**Comment by axcore** [comment]
>But unfortunately this option is not available for videos in a playlist (Download with name is greyed out).

There are ways around this; I would check the whole channel, then right-click the problem video and do **Download with name...**. Then I would press CTRL + A to select all videos, then I would click the problem video to remove it from the selection, then I would right-click the remaining selected videos to download them.

**Comment by axcore** [comment]
Should be fixed in v2.5.0.
